app/app/upload-non-cr3/page.tsx

```diff
@@ -0,0 +1,257 @@
+              </Col>
+            </Row>
+          )}
+          {data &&
+            !validationErrors &&
+            !uploadError &&
+            !success &&
+            !isMutating && (
+              <>
+                <Row>
+                  <Col>
+                    <Alert variant="info">
+                      <AlignedLabel>
+                        <FaCircleInfo className="me-2" />
+                        {`${data.length.toLocaleString()} records will be imported`}
+                      </AlignedLabel>
+                    </Alert>
+                  </Col>
+                </Row>
+                <Row>
+                  <Col>
+                    <Button
+                      role="submit"
+                      disabled={!data.length || isMutating}
+                      onClick={(e) => {
+                        e.preventDefault();
+                        onUpload();
+                      }}
+                    >
+                      Continue
+                    </Button>
+                  </Col>
+                </Row>
+              </>
+            )}
+          <Row className="mt-3">
+            <Col>
+              {parsing && (
+                <AlignedLabel>
+                  <Spinner variant="primary" className="me-2" />
+                  <span>Processing...</span>
+                </AlignedLabel>
+              )}
+              {!validationErrors && isMutating && (
+                <AlignedLabel>
+                  <Spinner variant="primary" className="me-2" />
+                  <span>Uploading records...</span>
+                </AlignedLabel>
+              )}
+              {success && (
+                <Alert variant="success">
+                  <AlignedLabel>
+                    <FaCircleCheck className="me-2" />
+                    <span>Records successfully imported</span>
+                  </AlignedLabel>
+                </Alert>
+              )}
+              {validationErrors && (
+                <Alert
+                  variant="danger"
+                  className="d-flex justify-content-between"
+                >
+                  <AlignedLabel>
+                    <FaTriangleExclamation className="me-2" />
+                    <span>
+                      Your file is invalid — please correct the below errors and
+                      try again.
+                    </span>
+                  </AlignedLabel>
+                </Alert>
+              )}
+              {uploadError && (
+                <Alert
+                  variant="danger"
+                  className="d-flex justify-content-between"
+                >
+                  <AlignedLabel>
+                    <FaTriangleExclamation className="me-2" />
+                    <span>
+                      There was an error uploading your file - please try again.
+                    </span>
+                  </AlignedLabel>
+                </Alert>
+              )}
+              {validationErrors && (
```

Comment:
the old version of this feature used a [fancy table](https://handsontable.com/) component that enabled the user to edit the CSV and re-submit it through the UI.

I opted to keep things simpler. AFAIK Xavier uses this feature four times per year.